watertap/unit_models/tests/test_coag_floc_zo_jartest.py

```diff
@@ -0,0 +1,419 @@
+                "property_package": model.fs.properties,
+                "chemical_additives": bad_dict2 })
+
+        bad_dict3 = {'Alum':
+                        {"parameter_data":
+                            {"mw_additive": (200, pyunits.g/pyunits.mol),
+                            "moles_salt_per_mole_additive": "foo-bar",
+                            "mw_salt": (100, pyunits.g/pyunits.mol)}
+                        }
+                     }
+        with pytest.raises(ConfigurationError):
+            model.fs.unit = CoagulationFlocculationZO_JarTestModel(default={
+                "property_package": model.fs.properties,
+                "chemical_additives": bad_dict3 })
+
+        bad_dict4 = {'Alum':
+                        {"parameter_data":
+                            {"mw_additive": (200, pyunits.g/pyunits.mol),
+                            "foo-bar": 3,
+                            "mw_salt": (100, pyunits.g/pyunits.mol)}
+                        }
+                     }
+        with pytest.raises(ConfigurationError):
+            model.fs.unit = CoagulationFlocculationZO_JarTestModel(default={
+                "property_package": model.fs.properties,
+                "chemical_additives": bad_dict4 })
+
+        bad_dict5 = {'Alum':
+                        {"parameter_data":
+                            {"mw_additive": (200, pyunits.g/pyunits.mol),
+                            "moles_salt_per_mole_additive": 3,
+                            "foo-bar": (100, pyunits.g/pyunits.mol)}
+                        }
+                     }
+        with pytest.raises(ConfigurationError):
+            model.fs.unit = CoagulationFlocculationZO_JarTestModel(default={
+                "property_package": model.fs.properties,
+                "chemical_additives": bad_dict5 })
+
+        bad_dict6 = {'Alum':
+                        {"parameter_data":
+                            {"mw_additive": "not a tuple",
+                            "moles_salt_per_mole_additive": 3,
+                            "mw_salt": (100, pyunits.g/pyunits.mol)}
+                        }
+                     }
+        with pytest.raises(ConfigurationError):
+            model.fs.unit = CoagulationFlocculationZO_JarTestModel(default={
+                "property_package": model.fs.properties,
+                "chemical_additives": bad_dict6 })
+
+# -----------------------------------------------------------------------------
+# Start test class with bad config
+class TestCoagulationZOJarTest_withBadProperties():
+    @pytest.fixture(scope="class")
+    def coag_obj_bad_properties(self):
+        model = ConcreteModel()
+        model.fs = FlowsheetBlock(default={"dynamic": False})
+
+        return model
+
+    @pytest.mark.unit
+    def test_build_model_catch_prop_errors(self, coag_obj_bad_properties):
+        model = coag_obj_bad_properties
+
+        with pytest.raises(ConfigurationError):
```

Comment:
Should there be or is there a message displayed to the user to let them know what was wrong with the config?

Reply:
Add match args string to make sure you catch the correct error.

Reply:
For an example, see lines 231-237 in https://github.com/watertap-org/watertap/blob/c43528fe33eff9c68bf64374072470980f6245c6/watertap/unit_models/zero_order/tests/test_anaerobic_mbr_mec_zo.py